lib/web_apis/web_apis.es6.js

```diff
@@ -0,0 +1,347 @@
+      args: [
+        {
+          name: 'browserName',
+          documentation: `The name of imported browser.`,
+          typeName: 'String',
+        },
+        {
+          name: 'browserVersion',
+          documentation: `The version of imported browser.`,
+          typeName: 'String',
+        },
+        {
+          name: 'OSName',
+          documentation: `The name of operating system.`,
+          typeName: 'String',
+        },
+        {
+          name: 'OSVersion',
+          documentation: `The version of operating system.`,
+          typeName: 'String',
+        },
+        {
+          name: 'apiCatalog',
+          documentation: `The web catalog of this version of
+            browser, tested on the given operating system and version.
+            The json object should be the JSON object returned from
+            com.web.catalog.apiExtractor`,
+          typeName: 'JSON',
+        },
+      ],
+      code: function(browserName, browserVersion,
+        OSName, OSVersion, apiCatalog) {
+        let browser = this.Browser.create({
+          browserName: browserName,
+          browserVersion: browserVersion,
+          OS: OSName,
+          OSVersion: OSVersion,
+        });
+        let browserId = browser.id;
+        this.browsers.put(browser);
+        let interfaceNames = Object.keys(apiCatalog);
+        for (let i = 0; i < interfaceNames.length; i += 1) {
+          let interfaceName = interfaceNames[i];
+          for (let j = 0; j < apiCatalog[interfaceName].length; j += 1) {
+            let APIName = apiCatalog[interfaceName][j];
+            let webInterface = this.WebInterface.create({
+              interfaceName,
+              APIName,
+            });
+            let interfaceId = webInterface.id;
+            this.interfaces.put(webInterface);
+            this.browserAPIs.put(this.BrowserAPI.create({
+              browserId,
+              interfaceId,
+              id: `${browserId}.${interfaceId}`,
+            }));
+          }
+        }
+      },
+    },
+    {
+      name: 'getBrowserKeys',
+      documentation: `Asynch function to return an array of stored
+        browsers' key. browserName and version are optional arguments
+        to filter keys.`,
+      args: [
+        {
+          name: 'browserName',
+          documentation: `The optional argument that the returned browser
+            key will match the given name`,
+          typeName: 'String',
+        },
+        {
+          name: 'version',
+          documentation: `The optional argument that the returned browser
+            key will match the given version`,
+          typeName: 'String',
+        },
+      ],
+      returns: {
+        typeName: 'StringArray',
+        documentation: `Return a string array contains matching browser keys`,
+      },
+      code: function(browserName, version) {
+        return new Promise((resolve) => {
+          this.browsers.select().then((arrayDAO) => {
```

Comment:
We're using a thenable, so I don't think we need `new Promise(...)`. Just `return this.browsers.select().then((arrayDAO) => { ... });`